lib/TaskProcessing/SummaryProvider.php

```diff
@@ -115,18 +126,23 @@ public function process(?string $userId, array $input, callable $reportProgress)
 		}
 
 		try {
+			$completions = [];
 			if ($this->openAiAPIService->isUsingOpenAi() || $this->openAiSettingsService->getChatEndpointEnabled()) {
-				$completion = $this->openAiAPIService->createChatCompletion($userId, $model, $prompt, null, null, 1, $maxTokens);
+				foreach ($prompts as $prompt) {
+					$completions[] = $this->openAiAPIService->createChatCompletion($userId, $model, $prompt, null, null, 1, $maxTokens);
+				}
 			} else {
-				$completion = $this->openAiAPIService->createCompletion($userId, $prompt, 1, $model, $maxTokens);
+				foreach ($prompts as $prompt) {
+					$completions[] = $this->openAiAPIService->createCompletion($userId, $prompt, 1, $model, $maxTokens);
+				}
 			}
 		} catch (Exception $e) {
 			throw new RuntimeException('OpenAI/LocalAI request failed: ' . $e->getMessage());
 		}
-		if (count($completion) > 0) {
+		if (array_reduce($completions, fn (bool $prev, array $next): bool => $prev && (count($next) > 0), true)) {
 			$endTime = time();
 			$this->openAiAPIService->updateExpTextProcessingTime($endTime - $startTime);
-			return ['output' => array_pop($completion)];
+			return ['output' => implode(' ', array_map('array_pop', $completions))];
```

Comment:
```suggestion
			$completionStrings = array_map('array_pop', $completions)
			return ['output' => implode(' ', $completionStrings)];
```
To fix the tests.

Reply:
Even so, I found that `array_map` now emits a warning if the callback function expects an argument passed by reference, which is the case for `array_pop`. I'll probably need to refactor the function to indirectly call `array_pop` (and probably replace it with `end` as that might be faster).

https://www.php.net/manual/en/function.array-map.php#refsect1-function.array-map-changelog